pytest-embedded-idf/pytest_embedded_idf/dut.py

```diff
@@ -127,3 +129,101 @@ def close(self) -> None:
         if not self.skip_check_coredump:
             self._check_coredump()
         super().close()
+
+    @dataclass
+    class Case:
+        index: int
+        name: str
+        type: str
+        keywords: List
+        groups: List
+        attributes: List
+        subcase: List
+
+    def parse_test_menu(
+        self, pattern="Here's the test menu, pick your combo:(.+)Enter test for running.", trigger: str = ''
+    ) -> List[Case]:
+        """
+        Get test case list from test menu via UART print.
+
+        This function should be called after `Press ENTER to see the list of tests` prompt is shown.
+
+        Args:
+            pattern: Pattern to match the output from chip, menu block should be in the first group.
+                     This will be directly passed to `pexpect.expect()`.
+            trigger: Keys to trigger chip to print test menu by UART.
+
+        Returns:
+            A `list` of avaliable test cases, in `Case`. `Case` has following fields:
+                - index: The index of the case, which can be used to run this case.
```

Comment:
It's better to move the descriptions into the class, and here could be simplified to "Returns: list of `Case`"

(this is to avoid modify two places if we're updating the class `Case`)